caffeine/core.py

```diff
@@ -120,16 +127,52 @@ def __attempt_autoactivation(self):
                 elif not self.get_activated():
                     logger.info("Fullscreen app detected. Inhibiting.")
 
-        if (process_running or fullscreen) and not self.__auto_activated:
+        # Let's look for playing audio:
+        self.music_procs = 0  # Number of supposed audio only streams. We can turn the screen off for those
+        screen_relevant_procs = 0  # Number of all audio streams including videos. We want to keep the screen on here
+
+        if not process_running and not fullscreen:
+            # Get all audio playback streams
+            audio_sinks = os.popen("pacmd list-sink-inputs").read()
+            for sink in audio_sinks.split("index:"):
+                is_running = False
+                role_music = False
+                for line in sink.split("\n"):
+                    if "state: RUNNING" in line:
+                        is_running = True
+                    if 'media.role = "music"' in line: # checked Spotify and Gnome Music. They use this attribute.
```

Comment:
I don't see this role being used in any of the apps I have at hand.
However, `state: RUNNING` is only true if an app is actively playing back sound, so it might be enough to check just for that.

Reply:
I found this role to be used by Spotify and Gnome Music. So it seems to me this is a way some pure music players announce themselves to pulseaudio. I use this property only to check if i can turn off the screen. If this property is not present, Caffeine will still kick in, but also keep the screen on. Therefore I am checking for both properties.